assets/js/components/GoogleChart/index.js

```diff
@@ -81,50 +83,22 @@ export default function GoogleChart( props ) {
 
 	const [ isChartLoaded, setIsChartLoaded ] = useState( false );
 
-	// Ensure we don't filter out columns that aren't data, but are things like
-	// tooltips or other content.
-	let nonDataColumns = [];
-	if ( data?.length ) {
-		nonDataColumns = data[ 0 ].reduce( ( acc, row, rowIndex ) => {
-			return row?.role ? [ ...acc, rowIndex ] : acc;
-		}, [] );
-	}
-
-	// If only certain columns should be displayed for the data set we have
-	// then filter out that data.
-	let modifiedData = data;
-	if ( selectedStats?.length > 0 ) {
-		modifiedData = data.map( ( row ) => {
-			return row.filter( ( _columnValue, columnIndex ) => {
-				return (
-					columnIndex === 0 ||
-					selectedStats.includes( columnIndex - 1 ) ||
-					nonDataColumns.includes( columnIndex - 1 )
-				);
-			} );
-		} );
-	}
+	const modifiedData = getGoogleChartData( data, selectedStats );
```

Comment:
I think it would be a bit clearer to rename this, along these lines, or maybe more explicitly like `getDataFilteredForStats`, what do you think?

```suggestion
	const filteredData = getFilteredChartData( data, selectedStats );
```